pkg/controller/petset/pet_set.go

```diff
@@ -348,7 +348,13 @@ func (psc *PetSetController) syncPetSet(ps *apps.PetSet, pets []*api.Pod) (int,
 			err = petManager.Delete(pet)
 		}
 		if err != nil {
-			it.errs = append(it.errs, err)
+			switch err.(type) {
+			case errUnhealthyPet:
+				// We are not passing this error up, but we don't increment numPets if we encounter it,
+				// since numPets directly translates to petset.status.replicas
+			default:
+				it.errs = append(it.errs, err)
```

Comment:
i wasn't asking for the continue, i was asking that you remove the err != nil block. does that makes sense? if you remove that *and* don't put in the `case nil`, you'll end up appending nil into the errs list. sorry for the back and forth.